code/obj/artifacts/artifact_items/handheld_storage.dm

```diff
@@ -0,0 +1,180 @@
+			src.wear_layer = MOB_BELT_LAYER
+		else
+			src.wear_image_icon = 'icons/mob/clothing/back.dmi'
+			src.icon_state = src.back_icons[artifact.artitype.name]
+			src.wear_layer = MOB_BACK_LAYER
+		..()
+
+	unequipped(mob/user)
+		var/datum/artifact/artifact = src.artifact
+		if (!artifact.activated)
+			return ..()
+
+		// reset worn effects
+		artifact.show_fx(src)
+		src.UpdateOverlays(src.wizard_gem_image, "gem")
+		src.wizard_gem_image = null
+		src.reset_visible_state()
+		..()
+
+	// reset state to pre-worn state
+	proc/reset_visible_state()
+		if (src.icon == initial(src.icon))
+			return
+
+		// reset icons and wear layer
+		src.icon = initial(src.icon)
+		src.icon_state = src.base_icon_state
+		src.item_state = src.base_item_state
+		src.wear_image_icon = initial(src.wear_image_icon)
+		src.wear_layer = initial(src.wear_layer)
+
+		// reapply overlay if needed
+		src.UpdateOverlays(src.wizard_gem_image, "gem")
+		src.wizard_gem_image = null
+
+	// used for martian bag of holding changing size
+	proc/martian_change_shape()
+		var/list/prev_contents = list()
+		for (var/atom/A as anything in src.storage?.get_contents())
+			prev_contents += A
+			src.storage.transfer_stored_item(A, get_turf(src))
+
+		src.create_storage(/datum/storage/artifact_bag_of_holding/martian, max_wclass = pick(W_CLASS_TINY, W_CLASS_SMALL, W_CLASS_NORMAL, W_CLASS_BULKY),
+			slots = rand(3, 13), opens_if_worn = TRUE)
+
+		for (var/atom/A as anything in prev_contents)
+			if (src.storage.check_can_hold(A) == STORAGE_CAN_HOLD)
+				src.storage.add_contents(A, null, FALSE)
+
+/datum/artifact/bag_of_holding
+	associated_object = /obj/item/artifact/bag_of_holding
+	type_name = "Bag of Holding"
+	rarity_weight = 200
+	validtypes = list("eldritch", "martian", "wizard")
+	validtriggers = list(/datum/artifact_trigger/force, /datum/artifact_trigger/electric, /datum/artifact_trigger/heat,
+		/datum/artifact_trigger/radiation, /datum/artifact_trigger/carbon_touch, /datum/artifact_trigger/silicon_touch,
+		/datum/artifact_trigger/cold)
+	type_size = ARTIFACT_SIZE_MEDIUM
+	react_xray = list(5, 91, 97, 11, "HOLLOW")
+
+	effect_activate(obj/O)
+		if (..())
+			return TRUE
+
+		var/obj/item/artifact/bag_of_holding/boh = O
+
+		switch(src.artitype.name)
+			// large variety of storage, items are stored in LIFO, FIFO, or random order
+			if ("eldritch")
+				var/slots = rand(3, 13)
+				var/wclass
+
+				// pick max w class that can be held
+				if (slots >= 3 && slots <= 5)
+					wclass = pick(prob(10); W_CLASS_TINY, prob(50); W_CLASS_SMALL, prob(100); W_CLASS_NORMAL, prob(40); W_CLASS_BULKY)
+				else if (slots > 5 && slots <= 9)
+					wclass = pick(prob(20); W_CLASS_TINY, prob(100); W_CLASS_SMALL, prob(50); W_CLASS_NORMAL)
+				else if (slots >= 9)
+					wclass = pick(prob(30); W_CLASS_TINY, prob(100); W_CLASS_SMALL, prob(10); W_CLASS_NORMAL)
+
+				// allow it to be worn if it can hold a lot of items
+				//if (slots > 4)
+				//	boh.c_flags |= ONBELT
```

Comment:
?

Reply:
Planned slot requirement for belt wearability, if it gets sprited, can remove if that'd be better